File: problem/Delete_Nodes/modifiedList.js
/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */
var modifiedList = function(nums, head) {
    let mpp = new Set(nums);
    
    while (head && mpp.has(head.val)) {
        head = head.next;
    }
    
    let curr = head;
    while (curr && curr.next) {
        while (curr.next && mpp.has(curr.next.val)) {
            curr.next = curr.next.next;
        }
        curr = curr.next;
    }
    
    return head;
};

File: problem/Delete_Nodes/modifiedList.c
struct ListNode *modifiedList(int *nums, int numsSize, struct ListNode *head)
{
    int *mpp = (int *)calloc(100001, sizeof(int));
    for (int i = 0; i < numsSize; i++)
    {
        mpp[nums[i]] = 1;
    }

    while (head && mpp[head->val])
    {
        head = head->next;
    }

    struct ListNode *curr = head;
    while (curr && curr->next)
    {
        while (curr->next && mpp[curr->next->val])
        {
            curr->next = curr->next->next;
        }
        curr = curr->next;
    }

    free(mpp);
    return head;
}